src/model/modelTrng_No_GPU.py

```diff
@@ -0,0 +1,300 @@
+        random_idx = get_random_idx(idx_master, batch_size)
+        imgs = hdf5_file["input"][random_idx,:]
+        imgs = imgs.reshape(input_shape)
+        imgs = np.swapaxes(imgs, 1, 3)
+
+        if not validation:  # Training need augmentation. Validation does not.
+            ## Need to augment
+            imgs = augment_data(imgs)
+
+        classes = hdf5_file["output"][random_idx, 0]
+
+        yield imgs, classes
+
+
+def get_idx_for_onesubset(hdf5_file, subset=0):
+    '''
+    Get the indices for one subset to be used in testing/validation
+    '''
+
+    idx_subset = np.where( (hdf5_file["subsets"][:,0] == int(subset)) )[0]
+
+    idx = {}
+    idx[0] = np.where( (hdf5_file['output'][idx_subset,0] == 0) )[0]
+    idx[1] = np.where( (hdf5_file['output'][idx_subset,0] == 1) )[0]
+
+    return idx
+
+def plot_loss_accuracy(mdl):
+
+    H = mdl.history
+    N = np.arange(0, len(H["loss"]))
+    plt.style.use("ggplot")
+    plt.figure()
+    plt.plot(N, H["loss"], label="train_loss")
+    plt.plot(N, H["val_loss"], label="test_loss")
+    plt.plot(N, H["acc"], label="train_acc")
+    plt.plot(N, H["val_acc"], label="test_acc")
+    plt.title(" Resnet3d Model Training on Volumetric data (No GPU's) ")
+    plt.xlabel("Epoch #")
+    plt.ylabel("Loss/Accuracy")
+    plt.legend()
+    plt.savefig("no_GPU_fig.png")
+    plt.close()
+
+#### MAIN  ######
+with h5py.File(path_to_hdf5, 'r') as hdf5_file:
+
+    print("Valid hdf5 file in 'read' mode: " + str(hdf5_file))
+    file_size = os.path.getsize(path_to_hdf5)
+    print('Size of hdf5 file: {:.3f} GB'.format(file_size/2.0**30))
+
+    num_rows = hdf5_file['input'].shape[0]
+    print("There are {} images in the dataset.".format(num_rows))
+    print('Class 0 count: {}'.format( len( np.where(hdf5_file["output"][:,0] == 0)[0] ) ) )
+    print('Class 1 count: {}'.format( len( np.where(hdf5_file["output"][:,0] == 1)[0] ) ) )
+
+    print("The datasets within the HDF5 file are:\n {}".format(list(hdf5_file.values())))
+
+    input_shape = tuple(list(hdf5_file["input"].attrs["lshape"]))
+    batch_size = int(BATCH_SIZE)
+    print ("Input shape of tensor = {}".format(input_shape))
+    print ("Batch Size  = {}".format(batch_size))
+
+    model = Resnet3DBuilder.build_resnet_18((32, 32, 32, 1), 1)  # (input tensor shape, number of outputs)
+    model.compile(optimizer='adam',loss='binary_crossentropy', metrics=['accuracy'])
+
+    tb_log = keras.callbacks.TensorBoard(log_dir=TB_LOG_DIR,
+                                histogram_freq=0,
+                                batch_size=batch_size,
+                                write_graph=True,
+                                write_grads=True,
+                                write_images=True,
+                                embeddings_freq=0,
+                                embeddings_layer_names=None,
+                                embeddings_metadata=None)
+
+    checkpointer = keras.callbacks.ModelCheckpoint(filepath=CHECKPOINT_FILENAME,
```

Comment:
ModelCheckpoint callback created but never used?

Reply:
Updated model fit generator to use checkpoint in callbacks list.
callbacks=[tb_log, checkpointer])